tests/shared.py

```diff
@@ -90,6 +91,48 @@ def get_mesh_for_testing(npts=None):
     return pybamm.Mesh(geometry, submesh_types, submesh_pts)
 
 
+def get_p2d_mesh_for_testing(npts=None, mpts=None):
+    param = pybamm.ParameterValues(
+        base_parameters={
+            "Negative electrode width": 0.3,
+            "Separator width": 0.2,
+            "Positive electrode width": 0.3,
+        }
+    )
+
+    geometry = pybamm.Geometry("1D macro", "1+1D micro")
+    param.process_geometry(geometry)
+
+    # provide mesh properties
+    submesh_types = {
+        "negative electrode": pybamm.Uniform1DSubMesh,
+        "separator": pybamm.Uniform1DSubMesh,
+        "positive electrode": pybamm.Uniform1DSubMesh,
+        "negative particle": pybamm.Uniform1DSubMesh,
+        "positive particle": pybamm.Uniform1DSubMesh,
+    }
+
+    if mpts is None:
+        submesh_pts = {
+            "negative electrode": {"x": 40},
+            "separator": {"x": 25},
+            "positive electrode": {"x": 35},
+            "negative particle": {"r": 10, "x": 40},
+            "positive particle": {"r": 10, "x": 35},
```

Comment:
this could cause trouble and needs to be changed or highlighted in a different issue. Does the negative particle need a value for x?

Reply:
I agree, we should really only be stating the pts for x once. Creating an issue.